src/main/java/teammates/logic/core/EmailGenerator.java

```diff
@@ -171,6 +171,34 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructorReminders(
     }
     
     /**
+     * Generates the feedback session closed emails for the given {@code session}.
+     * @throws EntityDoesNotExistException
+     */
+    public List<EmailWrapper> generateFeedbackSessionClosedEmails(FeedbackSessionAttributes session) {
+        
+        CourseAttributes course = coursesLogic.getCourse(session.getCourseId());
+        
+        List<InstructorAttributes> instructors = new ArrayList<InstructorAttributes>();
+        List<StudentAttributes> students = new ArrayList<StudentAttributes>();
+        if (!session.isPrivateSession()) {
```

Comment:
Deal with the unusual condition first. e.g.
```java
if (session.isPrivateSession()) {
    //return empty list
}
```

That way, much of the `normal flow' of the method remain un-indented. In particular, avoid cases where much of the method is inside a long `if` block.